### ISSUE TITLE
Import-Module Exchange or AzureAD Error in DevOps

### ISSUE THREAD

**Issue by jadamones** [microsoft/Microsoft365DSC#4382]
### Description of the issue

I'm using DevOps configured per the whitepaper for several tenants. Only one tenant has an issue where if I add the AzureAD and SPO composite resources at the same time, I get "The type initializer for 'Azure.Core.Pipeline.DiagnosticScopeFactory' threw an exception." only when running a compliance check. I can resolve this by putting both SPO and AAD configurations in the same composite resource under AzureAD. Then, if I add Exchange configuration to the Exchange resource, I get an error stating that the Exchange module needs to be imported when it is compiling the configuration. If I add the Exchange configurations to the Intune composite resource, it says Import-Module Intune, also, if I add Exchange configurations to AzureAD, it says Import-Module AzureAD. Obviously, I'd think it was an issue with the Exchange configuration (which I've done a fresh export of) but, if I leave the Exchange configurations in the resource, and then remove the AzureAD configurations, it will compile the Exchange configuration with no issues.

I have multiple tenants working with the exact same devops configuration that works as expected. I've also completely rebuilt this pipeline and it didn't resolve the problem. I'm struggling with where to even to start troubleshooting this and any guidance would be helpful. 
 

### Microsoft 365 DSC Version

1.24.221.1

### Which workloads are affected

Azure Active Directory, Exchange Online, SharePoint Online

### The DSC configuration

```powershell
@{
    AllNodes    = @(
        @{
            NodeName                    = 'localhost'
            CertificateFile             = '.\DSCNodeEncryption.cer'
            PsDscAllowPlainTextPassword = $true
            PsDscAllowDomainUser        = $true
        }
    )
    NonNodeData = @{
        Environment    = @{
            Name             = 'Production - Tenant Name'
            ShortName        = 'PRD-Tenant'
            TenantId         = '<tenantURL>'
            OrganizationName = '<tenantURL>'
        }
        Accounts       = @(
            @{
                Workload = 'Exchange'
                Account  = '<M365AdminAccount>'
            }
            @{
                Workload = 'Office365'
                Account  = '<M365AdminAccount>'
            }
            @{
                Workload = 'PowerPlatform'
                Account  = '<M365AdminAccount>'
            }
            @{
                Workload = 'SecurityCompliance'
                Account  = '<M365AdminAccount>'
            }
            @{
                Workload = 'SharePoint'
                Account  = '<M365AdminAccount>'
            }
            @{
                Workload = 'Teams'
                Account  = '<M365AdminAccount>'
            }
            @{
                Workload = 'Intune'
                Account  = '<M365AdminAccount>'
            }
            @{
                Workload = 'AzureAD'
                Account  = '<M365AdminAccount>'
            }
            @{
                Workload = 'OneDrive'
                Account  = '<M365AdminAccount>'
            }
        )
        AppCredentials = @(
            @{
                Workload       = 'Exchange'
                ApplicationId  = '<appid>'
                CertThumbprint = '<certhumb>'
            }
            @{
                Workload       = 'Office365'
                ApplicationId  = '<appid>'
                CertThumbprint = '<certhumb>'
            }
            @{
                Workload       = 'PowerPlatform'
                ApplicationId  = '<appid>'
                CertThumbprint = '<certhumb>'
            }
            @{
                Workload       = 'SecurityCompliance'
                ApplicationId  = '<appid>'
                CertThumbprint = '<certhumb>'
            }
            @{
                Workload       = 'SharePoint'
                ApplicationId  = '<appid>'
                CertThumbprint = '<certhumb>'
            }
            @{
                Workload       = 'Teams'
                ApplicationId  = '<appid>'
                CertThumbprint = '<certhumb>'
            }
            @{
                Workload       = 'Intune'
                ApplicationId  = '<appid>'
                CertThumbprint = '<certhumb>'
            }
            @{
                Workload       = 'AzureAD'
                ApplicationId  = '<appid>'
                CertThumbprint = '<certhumb>'
            }
            @{
                Workload       = 'OneDrive'
                ApplicationId  = '<appid>'
                CertThumbprint = '<certhumb>'
            }
        )
        }
    }
```

### Verbose logs showing the problem

```powershell
Compare-M365DSCConfigurations], [Confirm-M365DSCDependencies, Confirm-M365DSCDependencies], 
                    [Export-M365DSCConfiguration, Export-M365DSCConfiguration]...}
ExportedFunctions : {[Test-M365DSCAgent, Test-M365DSCAgent], [Set-M365DSCAgentCertificateConfiguration, 
                    Set-M365DSCAgentCertificateConfiguration], [Update-M365DSCResourceDocumentationPage, 
                    Update-M365DSCResourceDocumentationPage], [Save-M365DSCPartialExport, 
                    Save-M365DSCPartialExport]...}
ExportedVariables : {}
NestedModules     : {M365DSCAgent, M365DSCDocGenerator, M365DSCErrorHandler, M365DSCLogEngine...}

[2024-02-27 12:17:22] -     Required version: 1.24.221.1
[2024-02-27 12:17:22] -     Installed version: 1.24.221.1
[2024-02-27 12:17:22] -       Correct version installed, continuing.
[2024-02-27 12:17:22] - Modules installed successfully!
[2024-02-27 12:17:22] -  
[2024-02-27 12:17:22] - Preparing MOF compilation
[2024-02-27 12:17:22] -   Loading DSC configuration 'M365Configuration.ps1'
[2024-02-27 12:17:24] -   Preparing OutputFolder 'C:\agent\_work\4\s\Output'
[2024-02-27 12:17:24] -   Retrieving Credentials
[2024-02-27 12:17:24] -     Found 1 data file(s)
[2024-02-27 12:17:24] -       Processing: Production.psd1
[2024-02-27 12:17:24] -         Getting passwords from KeyVault 'M365DscVault'
[2024-02-27 12:17:24] -         Getting password from KeyVault for Exchange
[2024-02-27 12:17:25] -         Getting password from KeyVault for Office365
[2024-02-27 12:17:25] -         Getting password from KeyVault for PowerPlatform
[2024-02-27 12:17:25] -         Getting password from KeyVault for SecurityCompliance
[2024-02-27 12:17:25] -         Getting password from KeyVault for SharePoint
[2024-02-27 12:17:25] -         Getting password from KeyVault for Teams
[2024-02-27 12:17:26] -         Getting password from KeyVault for Intune
[2024-02-27 12:17:26] -         Getting password from KeyVault for AzureAD
[2024-02-27 12:17:26] -         Getting password from KeyVault for OneDrive
[2024-02-27 12:17:26] -  
[2024-02-27 12:17:26] - Start MOF compilation
[2024-02-27 12:17:26] -     Processing: Production.psd1
##[error]The module 'Exchange' could not be loaded. For more information, run 'Import-Module Exchange'.
##[error]PowerShell exited with code '1'.
Added TLS 1.2 in session.
```

**Comment by andikrueger** [comment]
Could you check if this issue is somehow related to this one: https://github.com/microsoftgraph/msgraph-sdk-powershell/issues/2478

**Comment by jadamones** [comment]
That would make some sense as to why moving the SPO configurations into the same composite resource and after the AAD configurations allows the compliance test to run! I was definitely able to replicate this on my agent VM using the issue that you linked.

Unfortuneately, I think I'm dealing with two different issues because that doesn't seem to affect the MOF compilation part once I add the Exchange configurations. I've noticed that there is an updated "Production.ps1" datafile in the devops script downloads. It has arrays for multiple Accepted Domains, DKIM, and Connectors defined. This tenant does have multiple accepted domains. Could that be part of the issue? I've confirmed that none of the deprecated parameters from the April breaking changes mentioned in[ this issue](https://github.com/microsoft/Microsoft365DSC/issues/3838) are present in the configuration. 

Thanks!
